disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+
+        self._answers: Dict[int, PollAnswer] = {}
+        for answer in answers:
+            if isinstance(answer, PollAnswer):
+                self._answers[len(self._answers) + 1] = answer
+            elif isinstance(answer, str):
+                self._answers[len(self._answers) + 1] = PollAnswer(PollMedia(answer))
+            else:
+                raise TypeError(
+                    f"Expected 'List[str]' or 'List[PollAnswer]' for 'answers', got List[{answer.__class__.__name__!r}]."
+                )
+
+        self.duration: Optional[timedelta] = duration
+        self.allow_multiselect: bool = allow_multiselect
+        self.layout_type: PollLayoutType = layout_type
+        self.is_finalized: bool = False
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} question={self.question} answers={self.answers!r}>"
+
+    @property
+    def answers(self) -> List[PollAnswer]:
+        """List[:class:`PollAnswer`]: return the list of answers for this poll."""
+        return list(self._answers.values())
+
+    @property
+    def expires_at(self) -> Optional[datetime]:
+        """Optional[:class:`datetime.datetime`]: the expiration date for this poll, if available."""
+        if not self.duration or not self.message:
+            return
+        return self.message.created_at + self.duration
+
+    def get_answer(self, answer_id: int, /) -> Optional[PollAnswer]:
+        """Return the requested poll answer.
+
+        Parameters
+        ----------
+        answer_id: :class:`int`
+            The answer id.
+
+        Returns
+        -------
+        Optional[:class:`PollAnswer`]
+            The requested answer.
+        """
+        return self._answers.get(answer_id)
+
+    @classmethod
+    def from_dict(
+        cls,
+        channel: MessageableChannel,
+        message: Message,
+        state: ConnectionState,
+        data: PollPayload,
+    ) -> Poll:
+        poll = cls(
+            question=PollMedia.from_dict(state, data["question"]),
+            answers=[],
+            allow_multiselect=data["allow_multiselect"],
+            layout_type=try_enum(PollLayoutType, data["layout_type"]),
+        )
+        for answer in data["answers"]:
+            poll._answers[int(answer["answer_id"])] = PollAnswer.from_dict(state, answer)
+
+        poll._state = state
+        poll.channel = channel
+        poll.message = message
+
+        # this should always be not None but we check
+        # for a null value anyway
+        if results := data.get("results"):
+            poll.is_finalized = results["is_finalized"]
+
+            for answer_count in results["answer_counts"]:
+                try:
+                    answer = poll._answers[int(answer_count["id"])]
+                except KeyError:
+                    # this should never happen
+                    continue
+                answer.count = answer_count["count"]
+                answer.self_voted = answer_count["me_voted"]
```

Comment:
not relevant to the rest of the review, but I really like how this is handled c:

Reply:
Ty 🙏🏻